src/amo/components/AddonsByAuthorsCard/index.js

```diff
@@ -51,64 +63,123 @@ type Props = {|
 
 export class AddonsByAuthorsCardBase extends React.Component<Props> {
   static defaultProps = {
+    pageParam: 'page',
+    paginate: false,
+    showMore: true,
     showSummary: false,
     type: 'horizontal',
-    showMore: true,
   }
 
   componentWillMount() {
-    const { addonType, authorUsernames, forAddonSlug } = this.props;
+    const {
+      addonType,
+      authorUsernames,
+      forAddonSlug,
+      location,
+      pageParam,
+      paginate,
+    } = this.props;
 
-    this.dispatchFetchAddonsByAuthors({ addonType, authorUsernames, forAddonSlug });
+    this.dispatchFetchAddonsByAuthors({
+      addonType,
+      authorUsernames,
+      forAddonSlug,
+      page: this.getCurrentPage({ location, paginate, pageParam }),
+    });
   }
 
   componentWillReceiveProps({
     addonType: newAddonType,
     authorUsernames: newAuthorNames,
     forAddonSlug: newForAddonSlug,
+    location: newLocation,
+    pageParam,
+    paginate,
   }: Props) {
     const {
       addonType: oldAddonType,
       authorUsernames: oldAuthorNames,
       forAddonSlug: oldForAddonSlug,
+      location: oldLocation,
     } = this.props;
 
+    const newPage = paginate ?
+      oldLocation.query[pageParam] !== newLocation.query[pageParam] : false;
```

Comment:
`newLocation` could be undefined, right? I'm not sure how to easily express that in a Flow type since it only applies to `componentWillReceiveProps`.

Reply:
I don't think it can be undefined since the `withRouter` HOC is used.

Reply:
Really? But when React calls `componentWillReceiveProps()` it does not go through any HOC as far as I know. I'm thinking of a situation where only partial props are updated, which I imagine could look like:

```js
instance.componentWillReceiveProps({ forAddonSlug: 'some-other-addon' })
```

Reply:
I learned that `componentWillReceiveProps()` receives the next `this.props` so it will be a complete object.